src/Workspaces/Core/Portable/ReassignedVariable/AbstractReassignedVariableService.cs

```diff
@@ -206,10 +208,20 @@ bool ComputeLocalIsAssigned(ILocalSymbol local)
                 if (local.DeclaringSyntaxReferences.Length == 0)
                     return false;
 
-                var localDeclaration = local.DeclaringSyntaxReferences.First().GetSyntax(cancellationToken);
-                if (localDeclaration.SyntaxTree != semanticModel.SyntaxTree)
+                var localDeclaration = local.DeclaringSyntaxReferences
+                    .Select(r => r.GetSyntax(cancellationToken))
+                    .Where(s => s.SyntaxTree == semanticModel.SyntaxTree)
+                    .FirstOrDefault();
+                if (localDeclaration == null)
                 {
-                    Contract.Fail("Local did not come from same file that we were analyzing?");
+                    try
+                    {
+                        throw new InvalidOperationException("Local did not come from same file that we were analyzing?");
+                    }
+                    catch (Exception ex) when (FatalError.ReportAndCatch(ex, ErrorSeverity.Critical))
+                    {
+                    }
```

Comment:
You shouldn't need to do the try/catch, we do that for you in the underlying machinery: https://github.com/dotnet/roslyn/blob/aeb9006e67cb50289d169a00c621ebbce7df9d0b/src/VisualStudio/Core/Def/Watson/WatsonExtensions.cs